cmake/Codegen.cmake

```diff
@@ -54,23 +54,16 @@ install(FILES ${CMAKE_BINARY_DIR}/caffe2/core/macros.h
 # ---[ ATen specific
 if (INTERN_BUILD_ATEN_OPS)
   IF(MSVC)
-	SET(OPT_FLAG 
-		$<$<CONFIG:Debug>:"${CMAKE_CXX_FLAGS_DEBUG}">
-		$<$<CONFIG:Release>:"${CMAKE_CXX_FLAGS_RELEASE}">
-		$<$<CONFIG:RelWithDebInfo>:"${CMAKE_CXX_FLAGS_RELWITHDEBINFO}">
-		$<$<CONFIG:MinSizeRel>:"${CMAKE_CXX_FLAGS_MINSIZEREL}">
-		" /fp:strict ")
-	SET(VCOMP_LIB "vcomp$<$<CONFIG:Debug>:${CMAKE_DEBUG_POSTFIX}>")
-	message(STATUS "OPT: ${OPT_FLAG}")
-	message(STATUS "COM: ${VCOMP_LIB}")
-  ELSE()
-	SET(OPT_FLAG "-O3 ")
-	SET(VCOMP_LIB "vcomp")
-	IF("${CMAKE_BUILD_TYPE}" MATCHES "Debug")
-	  SET(OPT_FLAG " ")
-	  SET(VCOMP_LIB "vcompd")
-	ENDIF()
-  ENDIF()
+    SET(OPT_FLAG "/fp:strict ")
+    SET(VCOMP_LIB "vcomp$<$<CONFIG:Debug>:${CMAKE_DEBUG_POSTFIX}>")
```

Comment:
We should not use `CMAKE_DEBUG_POSTFIX` here, because the lib is called `vcompd`. This variable is used to control how we name our libs.

Reply:
```
[2136/2940] cmd.exe /C "cd . && C:\Jenkins\Miniconda3\Library\bin\cmake.exe -E vs_link_dll --intdir=caffe2\CMakeFiles\torch_cpu.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100183~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100183~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~2\2017\BUILDT~1\VC\Tools\MSVC\1411~1.255\bin\HostX64\x64\link.exe  @CMakeFiles\torch_cpu.rsp  /out:bin\torch_cpu.dll /implib:LIB\torch_cpu.lib /pdb:bin\torch_cpu.pdb /dll /version:0.0 /machine:x64 /ignore:4049 /ignore:4217 /INCREMENTAL:NO /NODEFAULTLIB:vcomp$<$<CONFIG:Debug>:d>   && cd ."
FAILED: bin/torch_cpu.dll LIB/torch_cpu.lib 
cmd.exe /C "cd . && C:\Jenkins\Miniconda3\Library\bin\cmake.exe -E vs_link_dll --intdir=caffe2\CMakeFiles\torch_cpu.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100183~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100183~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~2\2017\BUILDT~1\VC\Tools\MSVC\1411~1.255\bin\HostX64\x64\link.exe  @CMakeFiles\torch_cpu.rsp  /out:bin\torch_cpu.dll /implib:LIB\torch_cpu.lib /pdb:bin\torch_cpu.pdb /dll /version:0.0 /machine:x64 /ignore:4049 /ignore:4217 /INCREMENTAL:NO /NODEFAULTLIB:vcomp$<$<CONFIG:Debug>:d>   && cd ."
&& was unexpected at this time.
```
Looks like it is not correctly used.

Reply:
The variable `VCOMP_LIB` is used like this in `cmake\public\utils.cmake`.
```
set_target_properties(${libname} PROPERTIES LINK_FLAGS_RELEASE "/NODEFAULTLIB:${VCOMP_LIB}")
set_target_properties(${libname} PROPERTIES LINK_FLAGS_DEBUG "/NODEFAULTLIB:${VCOMP_LIB}")
set_target_properties(${libname} PROPERTIES STATIC_LIBRARY_FLAGS "/NODEFAULTLIB:${VCOMP_LIB}")
```
Maybe we could make two variables `VCOMP_LIB` and `VCOMP_LIB_DEBUG` and then change the previous code into the following one.
```
set_target_properties(${libname} PROPERTIES LINK_FLAGS_RELEASE "/NODEFAULTLIB:${VCOMP_LIB}")
set_target_properties(${libname} PROPERTIES LINK_FLAGS_MINSIZEREL "/NODEFAULTLIB:${VCOMP_LIB}")
set_target_properties(${libname} PROPERTIES LINK_FLAGS_RELWITHDEBINFO "/NODEFAULTLIB:${VCOMP_LIB}")
set_target_properties(${libname} PROPERTIES LINK_FLAGS_DEBUG "/NODEFAULTLIB:${VCOMP_LIB_DEBUG}")

set_target_properties(${libname} PROPERTIES STATIC_LIBRARY_FLAGS_RELEASE "/NODEFAULTLIB:${VCOMP_LIB}")
set_target_properties(${libname} PROPERTIES STATIC_LIBRARY_FLAGS_MINSIZEREL "/NODEFAULTLIB:${VCOMP_LIB}")
set_target_properties(${libname} PROPERTIES STATIC_LIBRARY_FLAGS_RELWITHDEBINFO "/NODEFAULTLIB:${VCOMP_LIB}")
set_target_properties(${libname} PROPERTIES STATIC_LIBRARY_FLAGS_DEBUG "/NODEFAULTLIB:${VCOMP_LIB_DEBUG}")
```